lib/client/client.go

```diff
@@ -288,3 +316,95 @@ func (c *Client) list(metadata unversioned.ResourceMetadata, helper conversionHe
 
 	return nil
 }
+
+// ensureGlobalFelixConfig ensures the specific global felix configuration has
+// been initialized, setting to the default value if not.
+func (c *Client) ensureGlobalFelixConfig(key string, defaultValue interface{}) error {
+	kv := &model.KVPair{
+		Key:   model.GlobalConfigKey{Name: key},
+		Value: fmt.Sprintf("%v", defaultValue),
+	}
+	if _, err := c.backend.Create(kv); err == nil {
+		log.WithField(key, defaultValue).Info("Assigned default global Felix config")
+	} else {
+		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
+			log.WithField(key, defaultValue).WithError(err).Warnf("Failed to set global Felix config")
+			return err
+		}
+		log.WithField(key, defaultValue).Info("Using previously configured global Felix config")
+	}
+	return nil
+}
+
+// ensureGlobalFelixConfig ensures the specific global BGP configuration has
+// been initialized, setting to the default value if not.
+func (c *Client) ensureGlobalBGPConfig(key string, defaultValue interface{}) error {
+	kv := &model.KVPair{
+		Key:   model.GlobalBGPConfigKey{Name: key},
+		Value: fmt.Sprintf("%v", defaultValue),
+	}
+	if _, err := c.backend.Create(kv); err == nil {
+		log.WithField(key, defaultValue).Info("Assigned default global BGP config")
+	} else {
+		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
+			log.WithField(key, defaultValue).WithError(err).Warnf("Failed to set global BGP config")
+			return err
+		}
+		log.WithField(key, defaultValue).Info("Using previously configured global BGP config")
+	}
+	return nil
+}
+
+// ensureGlobalFelixInterfacePrefix ensures the global felix interface prefix
+// contains at least the prefix used by calico/node.
+func (c *Client) ensureGlobalFelixInterfacePrefix() error {
```

Comment:
It looks like Felix will default this, I think... https://github.com/projectcalico/felix/blob/c725a96a566f3116c419e48aa63c9ba6b6027541/go/felix/config/config_params.go#L116

Anyway, it feels pretty wrong to me that we set this here if this is `calico/node` specific.